pkg/services/pluginsintegration/clientmiddleware/tracing_middleware.go

```diff
@@ -29,17 +31,42 @@ type TracingMiddleware struct {
 // traceWrap returns a new context.Context which wraps a newly created span. The span will also contain attributes for
 // plugin id, org id and user login. The second function returned is a cleanup function, which should be called by the
 // caller (deferred) and will set the span status/error and end the span.
-func (m *TracingMiddleware) traceWrap(ctx context.Context, pluginContext backend.PluginContext, opName string) (context.Context, func(error)) {
+func (m *TracingMiddleware) traceWrap(
+	ctx context.Context, pluginContext backend.PluginContext, headers http.Header, opName string,
+) (context.Context, func(error)) {
 	// Start span
 	ctx, span := m.tracer.Start(ctx, "PluginClient."+opName)
 
 	// Attach some plugin context information to span
 	span.SetAttributes("plugin_id", pluginContext.PluginID, attribute.String("plugin_id", pluginContext.PluginID))
 	span.SetAttributes("org_id", pluginContext.OrgID, attribute.Int64("org_id", pluginContext.OrgID))
+	if settings := pluginContext.DataSourceInstanceSettings; settings != nil {
+		span.SetAttributes("datasource_name", settings.Name, attribute.Key("datasource_name").String(settings.Name))
+		span.SetAttributes("datasource_type", settings.Type, attribute.Key("datasource_type").String(settings.Type))
+		span.SetAttributes("datasource_uid", settings.UID, attribute.Key("datasource_uid").String(settings.UID))
+	}
 	if u := pluginContext.User; u != nil {
 		span.SetAttributes("user", u.Login, attribute.String("user", u.Login))
 	}
 
+	// Additional attributes from http headers
+	if len(headers) > 0 {
+		for _, h := range []struct {
+			header    string
+			attribute string
+		}{
+			{"X-Panel-Id", "panel_id"},
+			{"X-Dashboard-Id", "dashboard_id"},
```

Comment:
Just a quick note: `X-Dashboard-Id` is a legacy header and was changed to `X-Dashboard-Uid`, which is not an `int` but a `string`, a while ago.

Sorry for linking to the headers from ds_proxy previously, I think those are the relevant ones: https://github.com/grafana/grafana/blob/b1a7e1d33d6bc04432849be251c80deec984dd54/pkg/services/query/query.go#L26-L32